src/valiant/models/targeton.py

```diff
@@ -611,34 +667,29 @@ def get_pam_ext_ref_end(self, pos: int) -> int:
         )
 
     def get_sgrna_ids(self, spr: StrandedPositionRange) -> FrozenSet[str]:
-        return frozenset({
+        cds = frozenset({
             self._codon_to_pam_variant[codon_index].sgrna_id
```

Comment:
Method definition seems to be missing?

Reply:
The `_codon_to_pam_variant` attribute is set via `object.__setattr__`, so static analysis may fail to recognise it.